dipy/reconst/tests/test_dki_micro.py

```diff
@@ -0,0 +1,309 @@
+    assert_array_almost_equal(IDT[..., 2], RDIc, decimal=3)
+
+    # Check class object
+    wmtiM = dki_micro.KurtosisMicrostructureModel(gtab_2s, fit_method="WLS")
+    wmtiF = wmtiM.fit(DWIsim, mask=mask)
+    assert_almost_equal(wmtiF.awf, FIEc, decimal=3)
+    assert_almost_equal(wmtiF.axonal_diffusivity, ADIc, decimal=3)
+    assert_almost_equal(wmtiF.hindered_ad, ADEc, decimal=3)
+    assert_almost_equal(wmtiF.hindered_rd, RDEc, decimal=3)
+    assert_almost_equal(wmtiF.tortuosity, Torc, decimal=3)
+
+
+def test_dki_micro_predict_single_voxel():
+    # single fiber simulate (which is the assumption of our model)
+    fie = 0.49
+    ADi = 0.00099
+    ADe = 0.00226
+    RDi = 0
+    RDe = 0.00087
+
+    # prepare simulation:
+    theta = random.uniform(0, 180)
+    phi = random.uniform(0, 320)
+    angles = [(theta, phi), (theta, phi)]
+    mevals = np.array([[ADi, RDi, RDi], [ADe, RDe, RDe]])
+    frac = [fie*100, (1 - fie)*100]
+    signal, dt, kt = multi_tensor_dki(gtab_2s, mevals, angles=angles,
+                                      fractions=frac, snr=None)
+    signal_gt, da = multi_tensor(gtab_2s, mevals, angles=angles,
+                                 fractions=frac, snr=None)
+
+    # Defined DKI microstrutural model
+    dkiM = dki_micro.KurtosisMicrostructureModel(gtab_2s)
+
+    # Fit single voxel signal
+    dkiF = dkiM.fit(signal)
+
+    # Check predict of KurtosisMicrostruturalModel
+    pred = dkiM.predict(dkiF.model_params)
+    assert_array_almost_equal(pred, signal_gt)
+
+    pred = dkiM.predict(dkiF.model_params, S0=100)
+    assert_array_almost_equal(pred, signal_gt * 100)
+
+    # Check predict of KurtosisMicrostruturalFit
+    pred = dkiF.predict(gtab_2s, S0=100)
+    assert_array_almost_equal(pred, signal_gt * 100)
+
+
+def test_dki_micro_predict_multi_voxel():
+    dkiM = dki_micro.KurtosisMicrostructureModel(gtab_2s)
+    dkiF = dkiM.fit(DWIsim)
+
+    # Check predict of KurtosisMicrostruturalModel
+    pred = dkiM.predict(dkiF.model_params)
+    assert_array_almost_equal(pred, DWIsim_all_taylor, decimal=3)
+
+    pred = dkiM.predict(dkiF.model_params, S0=100)
+    assert_array_almost_equal(pred, DWIsim_all_taylor * 100, decimal=3)
+
+    # Check predict of KurtosisMicrostruturalFit
+    pred = dkiF.predict(gtab_2s, S0=100)
+    assert_array_almost_equal(pred, DWIsim_all_taylor * 100, decimal=3)
+
+
+def test_dki_micro_awf_only():
+    dkiM = dki_micro.KurtosisMicrostructureModel(gtab_2s)
+    dkiF = dkiM.fit(DWIsim, awf_only=True)
+    awf = dki_micro.axonal_water_fraction(dkiF.model_params)
+    assert_almost_equal(awf, FIE)
+
+
+def additional_tortuosity_tests():
+    # Test tortuosity when rd is zero
+    # single voxel
+    t = dki_micro.tortuosity(1.7e-3, 0.0)
+    assert_almost_equal(t, 0.0)
+
+    # multi-voxel
+    RDEc = RDE.copy()
+    Torc = Tor.copy()
+    RDEc[1, 1, 1] = 0.0
+    Torc[1, 1, 1] = 0.0
+    t = dki_micro.tortuosity(ADE, RDEc)
+    assert_almost_equal(Torc, t)
```

Comment:
I don't know if you saw my response to this comment at my original PR #1027, but testing this might be tricky since "assert_raises(ValueError, dkiF.hindered_evals)" generates the error before assert_raises evaluates it.